packages/playground/website/public/plugin-proxy.php

```diff
@@ -287,14 +312,43 @@ function ($curl, $body) use (&$extra_headers_sent, $default_headers) {
             $_GET['artifact']
         );
     } else if (isset($_GET['repo']) && isset($_GET['name'])) {
-
-      // Only allow downloads from the block-interactivity-experiments repo for now.
+        // Only allow downloads from the block-interactivity-experiments repo for now.
         if ($_GET['repo'] !== 'WordPress/block-interactivity-experiments') {
             throw new ApiException('Invalid repo. Only "WordPress/block-interactivity-experiments" is allowed.');
         }
 
         $downloader->streamFromGithubReleases($_GET['repo'], $_GET['name']);
+    } else if (isset($_GET['url'])) {
+        // Proxy the current request to $_GET['url'] and return the response,
+        // but only if the URL is allowlisted.
+        $url = $_GET['url'];
+        $allowed_domains = ['api.wordpress.org', 'w.org', 's.w.org'];
+        $parsed_url = parse_url($url);
+        if (!in_array($parsed_url['host'], $allowed_domains)) {
+            http_response_code(403);
+            echo "Error: The specified URL is not allowed.";
+            exit;
+        }
+
+        function get_all_headers()
+        {
+            $headers = [];
+            foreach ($_SERVER as $name => $value) {
+                if (substr($name, 0, 5) == 'HTTP_') {
+                    $name = str_replace(' ', '-', ucwords(str_replace('_', ' ', strtolower(substr($name, 5)))));
+                    $headers[$name] = $value;
+                }
```

Comment:
this code block needs explanation. it's doing things but not being very clear about what it wants to do. a good example in a comment would go far.

is it only meant for specific `HTTP_` headers, or is it meant for _any_ `HTTP_` headers? I'm not sure without looking it up what we would be wanting to modify here or why.

Reply:
I just updated it in [27ea39d](https://github.com/WordPress/wordpress-playground/pull/724/commits/27ea39dd9bb33aa6c295069cd1e0f14d27e2cba5) – what do you think now?